src/lpython/semantics/python_comptime_eval.h

```diff
@@ -96,24 +96,30 @@ struct PythonIntrinsicProcedures {
         if (args.size() != 1) {
             throw SemanticError("Intrinsic abs function accepts exactly 1 argument", loc);
         }
-        ASR::expr_t* trig_arg = args[0];
+        ASR::expr_t* arg = ASRUtils::expr_value(args[0]);
```

Comment:
Currently we evaluate all arguments and only if `value` is present, we call `eval_abs` on this value. Thus `ASRUtils::expr_value(args[0])` is equivalent to just `args[0]`.
```suggestion
        ASR::expr_t* arg = args[0];
```